.github/workflows/tests.yml

```diff
@@ -60,3 +60,163 @@ jobs:
           flags: tests
           env_vars: PYTHON
           name: ${{ matrix.python }}
+
+  cygwin:
+    runs-on: windows-latest
+    env:
+      FORCE_COLOR: true
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - '3.9'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Cygwin
+        uses: cygwin/cygwin-install-action@v2
+        with:
+          packages: >-
+            python39
+            python39-devel
+            python39-pip
+            python39-setuptools
+            cmake
+            gcc-core
+            gcc-g++
+            git
+            make
+            ninja
+
+      - name: Fix git dubious ownership
+        # This addresses the "fatal: detected dubious ownership in
+        # repository" and "fatal: not in a git directory" errors
+        # encountered when trying to run Cygwin git in a directory not
+        # owned by the current user. This happens when the tests run
+        # Cygwin git in a directory outside the Cygwin filesystem.
+        run: git config --global --add safe.directory '*'
+        shell: C:\cygwin\bin\env.exe CYGWIN_NOWINPATH=1 CHERE_INVOKING=1 C:\cygwin\bin\bash.exe -leo pipefail -o igncr {0}
+
+      - name: Get pip cache path
+        id: pip-cache-path
+        run: echo "path=$(cygpath -w $(python -m pip cache dir))" >> $GITHUB_OUTPUT
+        shell: C:\cygwin\bin\env.exe CYGWIN_NOWINPATH=1 CHERE_INVOKING=1 C:\cygwin\bin\bash.exe -leo pipefail -o igncr {0}
+
+      - name: Restore cache
+        # Cygwin Python cannot use binary wheels from PyPI. Building
+        # some dependencies takes considerable time. Caching the built
+        # wheels speeds up the CI job quite a bit.
```

Comment:
In that case the wheels will be rebuilt. That's why weels are tagged with a platform specifier. I don't foresee any problem with this. The effort to put in place and maintain this is negligible and it cuts the job time in half, which is quite nice when debugging.

Reply:
As long as it dramatically speeds up the CI at least one in five CI runs, it sounds like a huge win to me?

I'm quite sure that there will be cache hits more than 20% of the time though. :D

Reply:
My concern would be the new wheel don't get added to the cache; the next run continues to pull the old cache rather than getting a new version.  It seems the consensus is that the cache will be expired at least every couple of months, so this isn't really an issue.

Reply:
The cache is keyed with the hash of the commit that triggered the CI job but retrieved with a less specific key in LRU fashion, thus this is not a concern. You can read the documentation for `actions/cache` to get the details.

Reply:
That works.